File: leetcode/10. shuffleString/shuffleString.go
package shufflestring

import (
	"strings"
)

func restoreString(s string, indices []int) string {
	length := len(s)
	slist := make([]string, length)
	for i := 0; i < length; i++ {
		slist[indices[i]] = string(s[i])
	}
	return strings.Join(slist, "")
}

File: leetcode/10. shuffleString/shuffleString.py
from typing import List

def restoreString(s: str, indices: List[int]) -> str:
        length = len(s)
        slist = [""] * length
        for i in range(length):
            slist[indices[i]] = s[i]
        return "".join(slist)


def main():
    a = restoreString("abc", [1, 2, 0])
    print(a)
    
if __name__ == "__main__":
    main()